essence-of-live-coding/test/Cell/Util.hs

```diff
@@ -192,4 +212,98 @@ test =
           }
     ]
 
+type TestTraversables' = TestTraversables '[Maybe, [], V.Vector 10, Seq, Map Int]
+
+testTraverse' :: Test
+testTraverse' =
+  testGroup
+    "Traversing unit tests"
+    [ genTraversableTests' @TestTraversables' "traverse' (arr f) = arr (f <$>)" $
+        makeTraversableTest (traverseArrLaw @Int @Int)
+    , genTraversableTests' @TestTraversables'
+        "traverse' works as expected for any Cell Identy Int Int created with constructor Cell"
+        $ makeTraversableTest (traverseCellTest @Int @Int @Int)
+    , testProperty "traverse' by itself does not force the entire list (ArrM)" $
+        CellSimulation
+          { cell = arr head
+          , input = [1 : error "Bang !"]
+          , output = [1]
+          }
+    , testProperty "traverse' by itself does not force the entire list (Cell)" $
+        CellSimulation
+          { cell = cellId >>> arr head
+          , input = [1 : error "Bang !"]
+          , output = [1]
+          }
+    ]
+
+traverseArrLaw ::
+  forall a b t.
+  Traversable t =>
+  Proxy t ->
```

Comment:
Is it possible to do everything with `TypeApplications` instead of `Proxy`?

Reply:
If not, no worries.

Reply:
No I don't think it is. I don't think you can do partial application with type application, which I think would be needed...